apps/api/src/routers/user.py

```diff
@@ -67,13 +70,10 @@ async def me(
     return IdentityResponse(uid=user.uid, **user_record)
 
 
-@router.post("/apply", status_code=status.HTTP_201_CREATED)
-async def apply(
-    user: Annotated[User, Depends(require_user_identity)],
-    # media type should be automatically detected but seems like a bug as of now
-    raw_application_data: Annotated[
-        RawApplicationData, Form(media_type="multipart/form-data")
-    ],
+async def apply_flow(
+    user: User,
+    raw_application_data: Union[RawApplicationData],
```

Comment:
Question: Does the `raw_application_data` parameter need to be typed as a union? Is the plan to expand this in the future with the other application types, and if so, would that actually be necessary?

Reply:
I'm testing it taking other application types which I'm guessing is what it's supposed to do (correct me if I'm wrong Ian), and I'm running into the same issue as #452.

Reply:
Yes, the intention is to expand the union to include other application types, but using a discriminated union as you explained should be much better.

Reply:
Separate routes should be fine if they each have a concrete form model. Since this parameter isn't really used below before being dumped into `ProcessedApplicationData` where validation will happen a second time (assuming each route does a first time), I think this could just be a `dict` so you don't have to worry about the union.

Reply:
New thought: if `ProcessedApplicationData` is a [Type Adapter](https://docs.pydantic.dev/latest/concepts/type_adapter/) for a discriminated union, then it's okay to have `raw_application_data` typed as `Union[HackerApplicationData, MentorApplicationData, VolunteerApplicationData]` which allows any common pre-processing for special fields (e.g. resume), and then you can remove the second parameter for which type of application it is.